llvm/cmake/config-ix.cmake

```
@@ -294,6 +294,37 @@ if(LLVM_HAS_LOGF128)
   set(LLVM_HAS_LOGF128 "${HAS_LOGF128}")
 endif()
 
+# Check for ICU.
+if(LLVM_ENABLE_ICU)
+  set(LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES})
+  set(CMAKE_FIND_LIBRARY_SUFFIXES ".so")
```

@daltenty, do you happen to know if there is a way to specifically request a shared library on AIX using CMake (where shared libraries are often also archive files)?

---

I checked the AIX ICU libraries, this isn't a concern there as the OS only distributes the shared library version anyway. 

But this check should be generalized to account for the different suffixes on other platforms:
```suggestion
  set(CMAKE_FIND_LIBRARY_SUFFIXES "${CMAKE_SHARED_LIBRARY_SUFFIX}")
```

---

Thanks I've made this change